students/luke/s05/mailroom.py

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env python3
+
+"""
+https://uwpce-pythoncert.github.io/PythonCertDevel/exercises/mailroom.html#exercise-mailroom
+"""
+
+# Store donors/history in dictionary
+donor_list = {"alice": [5],
+              "bob": [1, 2],
+              "carol": [1, 5, 100],
+              "dan": [5],
+              "erin": [2, 1]}
+
+# Format of thank-you notes
+THANK_YOU = "Thank you, {}, for your donation(s) of ${}!\n"
+
+# Destination file
+OUTFILE = "{}.txt"
+
+
+def driver():
+    """ Main loop for mailroom """
+    done = False
+    while not done:
+        try:
+            action = input(("Select an action and press return:\n"
+                            "[S]end a thank-you (add users and donations)\n"
+                            "[W]rite thank-you letters to file\n"
+                            "[C]reate a report\n"
+                            "[Q]uit\n"
+                            "> ")).lower()
+            while action not in "swcq":
+                action = input("Bad option, reenter > ").lower()
+
+            if action == "s":
+                add_donations()
+            elif action == "w":
+                write_thankyous()
+            elif action == "c":
+                create_report()
+            else:
+                done = True
+        except EOFError:
```

Comment:
Curious where the EOFError would be coming from and why you are catching it without comment?